### DISCUSSION TITLE
Custom comparison functions

### DISCUSSION THREAD

**Discussion by SarloAkrobata** [moj-analytical-services/splink#1647] [General]
Hi, is it posible to write custom comparison function?
Like UDF in Spark.
Thanks.

**Comment by RobinL** [comment]
Yes, see here: https://moj-analytical-services.github.io/splink/topic_guides/comparisons/customising_comparisons.html?h=cust#method-4-providing-the-spec-as-a-dictionary

**Reply by SarloAkrobata** [reply]
This is on settings level, but what if I need to write custom python code, e.g I need to find diff for perceptual hashing or some custom scale price comparison?

**Reply by RobinL** [reply]
Not directly.  The function needs to be registered against whatever SQL backend you're using.

So if you can register the Python function as a UDF against your sql backend, then you can reference it in your SQL and it should work

**Comment by SarloAkrobata** [comment]
Can I write something like this:
```python
"comparisons": [
        my_really_custom_comparison_function("hash1", "hash2")
        ctl.name_comparison("first_name", term_frequency_adjustments_name=True),
        ctl.name_comparison("surname", term_frequency_adjustments_name=True),
        cl.levenshtein_at_thresholds("dob", [1, 2], term_frequency_adjustments=True),
        cl.levenshtein_at_thresholds("postcode_fake", 2, term_frequency_adjustments=True),
        cl.exact_match("birth_place", term_frequency_adjustments=True),
        cl.exact_match("occupation",  term_frequency_adjustments=True),
    ],```
